docs/design/proposals/intree-migrate.md

```diff
@@ -0,0 +1,106 @@
+they always have; however, behind the scenes Kubernetes hands control of all
+storage management operations (previously targeting in-tree drivers) to CSI
+drivers.
+
+The plan moving forward is to remove the in-tree plugins in kubernetes. CSI
+migration is the feature in k8s that can support a smooth transition from
+in-tree usage to CSI usage for volume related operations. Without CSI migration
+and if the in-tree driver is removed, It might cause some problem for users for
+their existing volume. The plan in kubernetes community is to eventually migrate
+all storage users to CSI. CSI migration will be an opt-in feature turned on at
+cluster creation time that will redirect in-tree plugin operations (Provision,
+Attach, Detach, Mount, Unmount(including Inline Volumes))
+to a corresponding CSI Driver. The migration library will redirect in-tree
+plugin usage to appropriate CSI drivers, while supporting seamless upgrade and
+downgrade between new Kubernetes version that uses CSI drivers for in-tree
+volume plugins to an old Kubernetes version that uses old-fashioned volume
+plugins without CSI, thus facilitates gradual roll-over to CSI
+
+## RBD in tree plugin to CSI migration
+
+This feature is tracked in kubernetes for v1.23 as alpha under
+[kubernetes/enhancements#2963](https://github.com/kubernetes/enhancements/pull/2963)
+. Below are the changes identified in the design discussions to enable CSI
+migration for RBD plugin:
+
+### ClusterID field in the migration request
+
+Considering the `clusterID` field is a required one for CSI, but in-tree SC
+has `monitors` field as a required thing the clusterID will be sent from the
+migration library based on the monitors field, kubernetes storage admin supposed
+to create a clusterID based on the monitors hash ( md5sum) in the CSI config map
+and keep the monitors under this configuration before enabling the migration.
+While CSI driver receive the volume handle it will look at the configmap and
+figure out the mons to do the operations. Thus, CSI operations are unchanged or
+untouched wrt to the clusterID field.
+
+### New Volume Handle/ID for existing volume operations
+
+The existing volume will have a volume handle passed in similar to the following
+format:
+
+```
+mig_mons-<monitors-hash>_image-<imageUID>_<pool-hash>
+```
+
+Details on the hash values:
+
+* Monitors Hash: this carry a hash value (md5sum) which will be acted as the
+  `clusterID` for the operations in this context.
+
+* ImageUID: this is the unique UUID generated by kubernetes for the created
+  volume.
+
+* PoolHash: this is an encoded string of pool name.
+
+The existing in-tree volume's node operations should work without any issues as
+those will be tracked as `static volumes` for the CSI driver. From above volume
+handle, the CSI driver can also connect to the backend cluster and clean/delete
+the image. The migration volume ID carry the information like mons, pool and
```

Comment:
>From above volume handle, the CSI driver can also connect to the backend cluster and clean/delete
the image. The migration volume ID

use volume ID not to mention both volume handle and volume ID? take care of this in other places also